lib/secrets/secrets.go

```diff
@@ -156,18 +179,37 @@ func (d *diskStore) Exists(secret SecretName) (bool, error) {
 }
 
 func (d *diskStore) GetSecret(secret SecretName) ([]byte, error) {
-	path := filepath.Join(d.basePath, secret.Group, secret.Name)
-	ret, err := ioutil.ReadFile(path)
+	var ret []byte
+
+	// checks if the encrypted secret exists
+	pathToEncryptedSecret := filepath.Join(d.basePath, secret.Group, secret.Name+".enc.toml")
+	encryptedSecretExists, err := fileutils.PathExists(pathToEncryptedSecret)
+
+	if encryptedSecretExists {
+		ret, err = ioutil.ReadFile(pathToEncryptedSecret)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not read secrets data (secret may not have been generated yet)")
+		}
+		// decrypts and returns the secret value
+		retE, err := getDecryptedData(d.basePath, ret)
+		if err != nil {
+			return nil, errors.Wrap(err, "could not decrypt secret with existing key)")
```

Comment:
There seems to be an unmatched parenthesis in the string

Reply:
fixed